server/service/confluence_service.go

```diff
@@ -0,0 +1,198 @@
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const ErrorStatusNotFound = "No content found"
+
+type ConfluenceStatus struct {
+	State string `json:"state"`
+}
+
+type ErrorResponse struct {
+	Message string `json:"message"`
+}
+
+func NormalizeConfluenceURL(confluenceURL string) (string, error) {
+	u, err := url.Parse(confluenceURL)
+	if err != nil {
+		return "", err
+	}
+
+	if u.Host == "" {
```

Comment:
Can you add a comment explaining the purpose of this code branch? Is this for URLs without scheme being parsed directly as `Path`?

Reply:
Under which situations does this happen?

Reply:
☝️

Reply:
@fmartingr 
this can be useful in all the cases when the URL is not provided in `https://github.com/mattermost-community` format
for example: `github.com/mattermost-community`

Reply:
Wouldn't it be better to validate that the URL contains an Scheme and Host rather than guessing from the Path?

Reply:
@fmartingr We are already checking for the scheme and host.
Added these extra checks to improve the UX if the user did not paste the complete URL but we can get the required data from whatever is provided by the user.